### PR TITLE
Updating to OWLAPI 4.5.22

### PR DESCRIPTION
**Pull request by hkir-dev**
Resolves [https://github.com/owlcs/owlapi/issues/1004]

- [ ] `docs/` have been added/updated
- [ ] tests have been added/updated
- [ ] `mvn verify` says all tests pass
- [ ] `mvn site` says all JavaDocs correct
- [ ] `CHANGELOG.md` has been updated

Changes Summary:
- Backward compatibility checking maven plugin (`com.github.siom79.japicmp`) `ignoreMissingClassesByRegularExpressions` parameter added in `robot-core/pom.xml`. Owlapi 4.5.22 uses a newer version of the rio-api: `sesame-rio-api-2.7.16.jar -> rdf4j-rio-api-2.4.2.jar.` Plugin is raising `"Could not load 'org.openrdf.rio.RDFHandlerException'"` exception, but that class renamed to `org.eclipse.rdf4j.rio.RDFHandlerException` now.
- Owlapi Manchester syntax parsing logic changed to [prevent illegal punning](https://github.com/owlcs/owlapi/blob/owlapi-parent-4.5.22/parsers/src/main/java/org/semanticweb/owlapi/manchestersyntax/parser/ManchesterOWLSyntaxParserImpl.java#L701). `QuotedEntityChecker.getOWLDataProperty`, `getOWLObjectProperty` and `getOWLClass` was causing illegal punning exception when called without a reference ontology. Added if controls to prevent illegal punning in QuotedEntityChecker.getOWLDataProperty and getOWLObjectProperty.
- Axiom annotations using undeclared annotation properties are now getting parsed in 4.5.22 (inferring type from usage), updated the test files accordingly. Example case: `oboInOwl:date_retrieved` not declared in [edit.owl](https://github.com/ontodev/robot/blob/owlapi-4.5.19/docs/examples/edit.owl)
- DLExpressivityChecker.Construct changed in the new owlapi, accordingly OntologyMetrics and MeasureOperationTest updated

### CODE REVIEW

File: docs/examples/example.obo [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
+owl-axioms: Prefix(owl:=<http://www.w3.org/2002/07/owl#>)\nPrefix(rdf:=<http://www.w3.org/1999/02/22-rdf-syntax-ns#>)\nPrefix(xml:=<http://www.w3.org/XML/1998/namespace>)\nPrefix(xsd:=<http://www.w3.org/2001/XMLSchema#>)\nPrefix(rdfs:=<http://www.w3.org/2000/01/rdf-schema#>)\n\n\nOntology(\nDeclaration(Class(<http://purl.obolibrary.org/obo/UBERON_0001062>))\n############################\n#   Classes\n############################\n\n# Class: <http://purl.obolibrary.org/obo/UBERON_0001062> (<http://purl.obolibrary.org/obo/UBERON_0001062>)\n\nSubClassOf(<http://purl.obolibrary.org/obo/UBERON_0001062> owl:Thing)\n\n\n)
```
Comment (matentzn) [inline docs/examples/example.obo:4]: Odd change if UBERON_0001062 is labelled, but ok. Nothing concerning.

File: docs/examples/example.obo [hunk 2]
```diff
@@ -307,20 +307,20 @@ xref: http://www.snomedbrowser.com/Codes/Details/278876000
...
...r for which there is no ligand Pocket. Understanding NR evolution has been further improved by comparison of several completed genomes, particularly those of deuterostomes and ecdysozoans. In contrast, evolution of NR ligands is still much debated. One hypothesis proposes that several independent gains and losses of ligand-binding ability in NRs occurred in protostomes and deuterostomes. A second hypothesis, pertaining to the NR3 subfamily (vertebrate steroid hormone receptors and estrogen related receptor), proposes that before the divergence of protostomes and deuterostomes, there was an ancestral steroid receptor (AncSR) that was ligand-activated and that orphan receptors secondarily lost the ability to bind a ligand. (...) Our analysis reveals that steroidogenesis has been independently elaborated in the 3 main bilaterian lineages (...).[well established][VHOG]" xsd:string {date_retrieved="2012-09-17", external_class="VHOG:0000098", ontology="VHOG", source="http://bgee.unil.ch/", source="DOI:10.1073/pnas.0812138106 Markov GV, Tavares R, Dauphin-Villemant C, Demeneix BA, Baker ME, Laudet V, Independent elaboration of steroid hormone signaling pathways in metazoans. PNAS (2009)"}
```
Comment (matentzn) [inline docs/examples/example.obo:317]: Som many annos were lost previously!

File: docs/examples/example.obo [hunk 3]
```diff
@@ -370,25 +371,25 @@ xref: http://www.snomedbrowser.com/Codes/Details/362584002
...
+xref: UMLS:C0001613 {source="ncithesaurus:Adrenal_Cortex"}
 xref: VHOG:0001481
 xref: XAO:0000165
 is_a: UBERON:0001851 ! cortex
 intersection_of: UBERON:0001851 ! cortex
 intersection_of: part_of UBERON:0002369 ! adrenal gland
-property_value: depicted:by http://upload.wikimedia.org/wikipedia/commons/5/5c/Gray1185.png
-property_value: depicted:by http://upload.wikimedia.org/wikipedia/commons/thumb/5/5c/Gray1185.png/200px-Gray1185.png
-property_value: homology_notes "All craniates have groups of cells homologous to the mammalian adrenocortical and chromaffin tissues (medulla), but they are scattered in and near the kidneys in fishes. (...) The cortical and chromaffin tissues come together to form adrenal glands in tetrapods.[well established][VHOG]" xsd:string
+property_value: depicted:by "http://upload.wikimedia.org/wikipedia/commons/5/5c/Gray1185.png"
```
Comment (matentzn) [inline docs/examples/example.obo:380]: Interesting that literals are now wrapped in ""

@hkir-dev is this backwards compatible, i.e. a non-wrapped IRI (OBO format) can still be parsed by the new owlapi?

File: docs/examples/metrics_all.json [hunk 4]
```diff
@@ -30,8 +30,8 @@
     "dt_builtin_count_incl": 2,
     "dt_notbuiltin_count": 0,
     "dt_notbuiltin_count_incl": 0,
-    "expressivity": "SRIQ",
-    "expressivity_incl": "SRIQ",
+    "expressivity": "UCINTERIQ",
```
Comment (matentzn) [inline docs/examples/metrics_all.json:33]: What on earth is `UCINETERIQ`???? cc @ignazio1977

File: robot-core/pom.xml [hunk 5]
```diff
@@ -33,50 +33,50 @@
...
+<!--         </configuration> -->
+<!--         <executions> -->
+<!--           <execution> -->
+<!--             <phase>verify</phase> -->
+<!--             <goals> -->
+<!--               <goal>cmp</goal> -->
+<!--             </goals> -->
+<!--           </execution> -->
+<!--         </executions> -->
+<!--       </plugin> -->
```
Comment (matentzn) [inline robot-core/pom.xml:71]: Can we get some comments on the necessity to comment out the above?

File: robot-core/src/main/java/org/obolibrary/robot/QueryOperation.java [hunk 6]
```diff
@@ -17,14 +39,18 @@
 import org.apache.jena.sparql.core.DatasetGraph;
 import org.apache.jena.sparql.core.DatasetGraphFactory;
 import org.apache.jena.update.UpdateAction;
-import org.openrdf.model.BNode;
-import org.openrdf.model.Literal;
-import org.openrdf.model.Statement;
-import org.openrdf.model.URI;
-import org.openrdf.rio.RDFHandler;
+import org.eclipse.rdf4j.model.BNode;
```
Comment (matentzn) [inline robot-core/src/main/java/org/obolibrary/robot/QueryOperation.java:42]: Of note to other reviewers, not concerned, just noting.

File: robot-core/src/main/java/org/obolibrary/robot/QuotedEntityChecker.java [hunk 7]
```diff
@@ -389,10 +389,15 @@ public OWLDataProperty getOWLDataProperty(@Nonnull String name) {
...
-        return dataFactory.getOWLDataProperty(iri);
+    // prevent punning
+    if (!objectProperties.containsKey(name)) {
+      if (ioHelper != null) {
+        iri = ioHelper.createIRI(name);
+        if (iri != null) {
+          OWLDataProperty owlDataProperty = dataFactory.getOWLDataProperty(iri);
+          dataProperties.put(name, iri);
+          return owlDataProperty;
+        }
```
Comment (matentzn) [inline robot-core/src/main/java/org/obolibrary/robot/QuotedEntityChecker.java:400]: Hm, can we link this to a ticket, not sure what this change is/means.

File: robot-core/src/test/java/org/obolibrary/robot/MeasureOperationTest.java [hunk 8]
```diff
@@ -47,7 +48,7 @@ public void testExtendedMetrics() throws IOException {
     MeasureResult result =
         MeasureOperation.getMetrics(
             ontology, "extended", new CURIEShortFormProvider(new IOHelper().getPrefixes()));
-    assertEquals("SI(D)", result.getSimpleMetricValue(MetricsLabels.EXPRESSIVITY));
+    assertEquals("RRESTRUCINTE+I(D)", result.getSimpleMetricValue(MetricsLabels.EXPRESSIVITY));
```
Comment (matentzn) [inline robot-core/src/test/java/org/obolibrary/robot/MeasureOperationTest.java:51]: 😱

**Review by matentzn — COMMENTED** [review]
AWESOME this is absolutely amazing. THANK you so much @hkir-dev!!!

File: docs/examples/example.obo (hunk 3 above)
Comment (hkir-dev) [reply docs/examples/example.obo:380]: yes, it is backward compatible. Not quoted and quoted IRIs produced the same ontology.

File: docs/examples/metrics_all.json (hunk 4 above)
Comment (hkir-dev) [reply docs/examples/metrics_all.json:33]: Probably coming from [here](https://github.com/owlcs/owlapi/blob/owlapi-parent-4.5.22/api/src/main/java/org/semanticweb/owlapi/util/Construct.java)

File: robot-core/src/main/java/org/obolibrary/robot/QuotedEntityChecker.java (hunk 7 above)
Comment (hkir-dev) [reply robot-core/src/main/java/org/obolibrary/robot/QuotedEntityChecker.java:400]: Related with: https://github.com/owlcs/owlapi/issues/851
> Property punning between properties is illegal, but punning of a property name and class name or individual name is allowed

File: robot-core/src/test/java/org/obolibrary/robot/MeasureOperationTest.java (hunk 8 above)
Comment (hkir-dev) [reply robot-core/src/test/java/org/obolibrary/robot/MeasureOperationTest.java:51]: :)  [Construct.java](https://github.com/owlcs/owlapi/blob/owlapi-parent-4.5.22/api/src/main/java/org/semanticweb/owlapi/util/Construct.java)
ROLE_DOMAIN_RANGE + CONCEPT_UNION + CONCEPT_INTERSECTION + FULL_EXISTENTIAL + ROLE_TRANSITIVE + Use of data properties

File: robot-core/pom.xml (hunk 5 above)
Comment (hkir-dev) [reply robot-core/pom.xml:71]: Seems I can enable plugin and use this filter:
```Xml
			<ignoreMissingClassesByRegularExpressions>
				     <ignoreMissingClassesByRegularExpression>org.openrdf.rio.*</ignoreMissingClassesByRegularExpression>
			</ignoreMissingClassesByRegularExpressions>
```
Updating the pom accordingly.

File: docs/examples/metrics_all.json (hunk 4 above)
Comment (ignazio1977) [reply docs/examples/metrics_all.json:33]: Yep, CINT is for concept intersection. There is an outstanding issue with that class, the weird naming might depend on it.

Comment (matentzn) [reply docs/examples/metrics_all.json:33]: If I tell people my ontology is written in a language called UCINETERIQ they will not use it 🤣

**Comment by hkir-dev** [comment]
Applying all changes in a new PR: https://github.com/ontodev/robot/pull/1005/

**Pull request closed without merging**